### PR TITLE
Javascript javascript1 week3

### PR DESCRIPTION
**Pull request by chunmeishui**
finished js week 3 homework

### CODE REVIEW

**Review by amritpandey — COMMENTED** [review]
ohh chunmei you did really great job. Specially I like your  two unique features in question 4 Note taking app to check the existence of ID.

File: javascript/javascript1/week3/index.js [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+        title: "Game of thrones",
+        days: 3,
+        hours: 1,
+        minutes: 0,
+    }
+];
+
+function logOutSeriesText(seriesDurations) {
+    // write code here
+    let string = "";
```
Comment (aina21) [inline javascript/javascript1/week3/index.js]: string is not a good name for the variable as it's a keyword, maybe result or outputText

File: javascript/javascript1/week3/index.js [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+saveNote("Pick up groceries", 1);
+saveNote("Do laundry", 2);
+
+console.log(notes); // [{content: 'Pick up groceries', id: 1}, {content: 'Do laundry', id: 2}]
+
+
+//Get a note------------
+function getNote(id) {
+    // get the object with name of id
+    for (const getId of notes) {
```
Comment (aina21) [inline javascript/javascript1/week3/index.js:125]: I think note or Anote is a better naming than getId.It's better to start function naming with the verb. you could check this link for more info:
[https://javascript.plainenglish.io/javascript-naming-convention-best-practices-b2065694b7d](url)

File: javascript/javascript1/week3/index.js [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+// The note with id: 2, has the following note text: Do laundry
+
+
+
+//Unique feature-------------
+//judge the same id ,if already have that id, we can not add same id.
+function saveNote(content, id) {
+    let idExistence = false;
+    //judge the activities is true or false
+    for (const checkId of notes) {
```
Comment (aina21) [inline javascript/javascript1/week3/index.js:157]: same here

File: javascript/javascript1/week3/index.js [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+        if (id === checkId.id) {
+            idExistence = true;
+            break;
+        }
+    }
+    //judge two codition true or false----
+    if (idExistence === false) {
```
Comment (aina21) [inline javascript/javascript1/week3/index.js:164]: if (!idExistence)

File: javascript/javascript1/week3/index.js [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+        notes.push({
+            "content": content,
+            "id": id
+        })
+        return "this id is added successfully";
+    }
+    else {
```
Comment (aina21) [inline javascript/javascript1/week3/index.js:171]: as you use return on if, you don't need else

**Review by aina21 — APPROVED** [review]
good job :+1:

File: javascript/javascript1/week3/index.js (hunk 1 above)
Comment (chunmeishui) [reply javascript/javascript1/week3/index.js]: oh, yes you are right.

File: javascript/javascript1/week3/index.js (hunk 2 above)
Comment (chunmeishui) [reply javascript/javascript1/week3/index.js:125]: I can not open the link.

**Comment by chunmeishui** [comment]
> ohh chunmei you did really great job. Specially I like your two unique features in question 4 Note taking app to check the existence of ID.

File: javascript/javascript1/week3/index.js (hunk 1 above)
Comment (chunmeishui) [reply javascript/javascript1/week3/index.js]: Hi aina, I have changed all of the issues. pls, check for me again.

**Comment by chunmeishui** [comment]
Thanks for your encourage and help . Sent from Mail for Windows From: amritpandeySent: Friday, 12 November 2021 22.03To: chunmeishui/hyf-homeworkCc: chunmei; AuthorSubject: Re: [chunmeishui/hyf-homework] Javascript javascript1 week3 (PR ***@***.*** commented on this pull request.ohh chunmei you did really great job. Specially I like your two unique features in question 4 Note taking app to check the existence of ID.—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.

**Pull request merged**